avx512_gemm.h

```diff
@@ -1,5 +1,9 @@
 #pragma once
 
+#include "config.h"
+
+#ifdef INTGEMM_COMPILER_SUPPORTS_AVX512
```

Comment:
In general I like positive defines.  I'm a bit worried that somebody will include the code in another package, forget to define this, and not like the performance.  The nice thing about a negative define is it breaks the build if somebody screws up the defines.

Reply:
So maybe we should change this define and checking if it's true/false rather then defined/undefined and if it's undefined then just throw an error.